### ISSUE TITLE
New Check WhitespaceAroundArrayInitialization

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#9346]
Continuing #3203 
Code:-
```
class T {
  int[] arr = {1,2};
}
```
Config:-
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
    <module name="WhitespaceAround">
    
      <property name="tokens" value="ARRAY_INIT,RCURLY"/>
    </module>
    </module>
</module>
```
The code above with config specified does not raise any violation.
After a few trials, it was decided to make a new check for this. https://github.com/checkstyle/checkstyle/pull/9239#issuecomment-787075583

To resolve this issue, we should:

1. Remove ARRAY_INIT token from WhitespaceAround
2. Add some documentation to WhitespaceAround that this check does not verify whitespace around array initialization, citing https://google.github.io/styleguide/javaguide.html#s4.8.3.1-array-initializers

**Comment by Vyom-Yadav** [comment]
@romani Kindly approve it.

**Comment by Vyom-Yadav** [comment]
@romani There is no response from anybody, shall I close this? Then you can also close #3203

**Comment by romani** [comment]
Please show CLI execution and it's output

**Comment by Vyom-Yadav** [comment]
@romani CLI execution and its output is there now.

**Comment by Vyom-Yadav** [comment]
@romani Is this issue approved? Can I work on it?

**Comment by Vyom-Yadav** [comment]
@romani ping, can you approve this? Other members suggested this as a solution in https://github.com/checkstyle/checkstyle/pull/9239#issuecomment-787075583 I was working on it for some time so I wanted to complete this.
